sqlalchemy-stubs/ext/asyncio/engine.pyi

```diff
@@ -1,103 +1,128 @@
-    sync_connection: Any = ...
+    sync_connection: Optional[Connection] = ...
     def __init__(
         self,
         async_engine: AsyncEngine,
         sync_connection: Optional[Connection] = ...,
     ) -> None: ...
-    async def start(self): ...
+    async def start(self: _TAsyncConnection) -> _TAsyncConnection: ...
     @property
-    def connection(self) -> None: ...
-    async def get_raw_connection(self): ...
+    def connection(self) -> NoReturn: ...
+    async def get_raw_connection(self) -> Any: ...
     @property
-    def info(self): ...
+    def info(self) -> Mapping[Any, Any]: ...
     def begin(self) -> AsyncTransaction: ...
     def begin_nested(self) -> AsyncTransaction: ...
-    async def invalidate(self, exception: Optional[Any] = ...): ...
-    async def get_isolation_level(self): ...
-    async def set_isolation_level(self): ...
-    def in_transaction(self): ...
-    def in_nested_transaction(self): ...
-    def get_transaction(self): ...
-    def get_nested_transaction(self): ...
-    async def execution_options(self, **opt: Any): ...
+    async def invalidate(self, exception: Optional[Any] = ...) -> None: ...
+    async def get_isolation_level(self) -> Any: ...
+    async def set_isolation_level(self) -> Any: ...
+    def in_transaction(self) -> bool: ...
+    def in_nested_transaction(self) -> bool: ...
+    def get_transaction(self) -> Optional[AsyncTransaction]: ...
+    def get_nested_transaction(self) -> Optional[AsyncTransaction]: ...
+    async def execution_options(
+        self: _TAsyncConnection, **opt: Any
+    ) -> _TAsyncConnection: ...
     async def commit(self) -> None: ...
     async def rollback(self) -> None: ...
     async def close(self) -> None: ...
     async def exec_driver_sql(
         self,
         statement: Executable,
-        parameters: Optional[Mapping] = ...,
-        execution_options: Mapping = ...,
+        parameters: Optional[Mapping[Any, Any]] = ...,
+        execution_options: Mapping[Any, Any] = ...,
     ) -> Result: ...
     async def stream(
         self,
         statement: Executable,
-        parameters: Optional[Mapping] = ...,
-        execution_options: Mapping = ...,
+        parameters: Optional[Mapping[Any, Any]] = ...,
+        execution_options: Mapping[Any, Any] = ...,
     ) -> AsyncResult: ...
     async def execute(
         self,
         statement: Executable,
-        parameters: Optional[Mapping] = ...,
-        execution_options: Mapping = ...,
+        parameters: Optional[Mapping[Any, Any]] = ...,
+        execution_options: Mapping[Any, Any] = ...,
     ) -> Result: ...
     async def scalar(
         self,
         statement: Executable,
-        parameters: Optional[Mapping] = ...,
-        execution_options: Mapping = ...,
+        parameters: Optional[Mapping[Any, Any]] = ...,
+        execution_options: Mapping[Any, Any] = ...,
+    ) -> Any: ...
+    async def run_sync(
+        self, fn: Callable[..., Any], *arg: Any, **kw: Any
```

Comment:
the first element of the callable is a `Connection`

Reply:
Unfortunately, I'm not sure we can reliably type that at this time

Reply:
`Callable[[sometype, ...], ...]` is not a thing I guess

Reply:
No, it's not